File: frontend/src/types/filingSubmission.ts
export interface SubmissionValidationIssue {
  field_key: string;
  message: string;
  severity: string;
}

export interface FilingSubmissionPayload {
  case_type?: string | null;
  list_type?: string | null;
  with_application: boolean;

  petitioner_party_type?: string | null;
  petitioner_name?: string | null;
  hide_party_petitioner: boolean;
  differently_abled_petitioner: boolean;

  respondent_party_type?: string | null;
  respondent_name?: string | null;
  hide_party_respondent: boolean;
  differently_abled_respondent: boolean;

  advocates: Array<Record<string, string | null>>;
}

export interface FilingSubmissionPreviewResponse {
  document_id: number;
  extraction_job_id?: number | null;
  reviewed_session_id?: number | null;
  target_system: string;
  dry_run: boolean;
  ready_for_submit: boolean;
  payload: FilingSubmissionPayload;
  validation_issues: SubmissionValidationIssue[];
  audit_meta: Record<string, unknown>;
}

export interface DryRunSubmissionResponse {
  document_id: number;
  target_system: string;
  dry_run: boolean;
  ready_for_submit: boolean;
  validation_issues: SubmissionValidationIssue[];
  automation_result: Record<string, unknown>;
}

File: backend/app/schemas/filing_submission.py
from __future__ import annotations

from pydantic import BaseModel, Field


class SubmissionValidationIssue(BaseModel):
    field_key: str
    message: str
    severity: str = "error"


class FilingSubmissionPayload(BaseModel):
    case_type: str | None = None
    list_type: str | None = None
    with_application: bool = False

    petitioner_party_type: str | None = None
    petitioner_name: str | None = None
    hide_party_petitioner: bool = False
    differently_abled_petitioner: bool = False

    respondent_party_type: str | None = None
    respondent_name: str | None = None
    hide_party_respondent: bool = False
    differently_abled_respondent: bool = False

    advocates: list[dict] = Field(default_factory=list)
    full_metadata: dict = Field(default_factory=dict)


class FilingSubmissionPreviewResponse(BaseModel):
    document_id: int
    extraction_job_id: int | None = None
    reviewed_session_id: int | None = None
    target_system: str = "phhc_filing"
    dry_run: bool = True
    ready_for_submit: bool = False
    payload: FilingSubmissionPayload
    validation_issues: list[SubmissionValidationIssue] = Field(default_factory=list)
    audit_meta: dict = Field(default_factory=dict)


class PrepareSubmissionRequest(BaseModel):
    dry_run: bool = True


class DryRunSubmissionResponse(BaseModel):
    document_id: int
    target_system: str
    dry_run: bool = True
    ready_for_submit: bool
    validation_issues: list[SubmissionValidationIssue] = Field(default_factory=list)
    automation_result: dict = Field(default_factory=dict)
